### PR TITLE
[Chips] Exposing color for surface of chip view. …

### PR DESCRIPTION
**Pull request by mohammadcazig**
Pivotal: https://www.pivotaltracker.com/story/show/156742616
This is different than background color and sits on top of background color. 
We are required to have these two color properties to theme the component properly.
Chip color can sit on top of expected background color with specific opacity
screen shot 2018-04-12 at 4 00 52 pm
screen shot 2018-04-12 at 4 02 34 pm

### CODE REVIEW

File: components/Chips/src/MDCChipView.h [hunk 1]
```diff
@@ -161,6 +161,27 @@
...
+
+ @param state The control state.
+ @return The chip color.
+ */
+- (nullable UIColor *)chipColorForState:(UIControlState)state;
+
+/*
+ A color used as the chip's @c color.
+
+ Defaults to blue.
```
Comment (jverkoey) [inline components/Chips/src/MDCChipView.h]: This may be a lie?

File: components/Chips/src/MDCChipView.m [hunk 2]
```diff
@@ -609,6 +638,13 @@ - (void)layoutSubviews {
         [UIBezierPath bezierPathWithRoundedRect:self.bounds cornerRadius:cornerRadius].CGPath;
   }
 
+  CAShapeLayer *shapedLayer = (CAShapeLayer *)_chipView.layer.mask;
+  if (![shapedLayer isKindOfClass:[CAShapeLayer class]]) {
+    shapedLayer = [CAShapeLayer layer];
+    _chipView.layer.mask = shapedLayer;
+  }
+  shapedLayer.path = self.layer.shadowPath;
```
Comment (jverkoey) [inline components/Chips/src/MDCChipView.m]: Was this intentionally added? It doesn't look relevant to the change.

File: components/Chips/src/MDCChipView.h [hunk 3]
```diff
@@ -161,6 +161,27 @@
 - (void)setBackgroundColor:(nullable UIColor *)backgroundColor forState:(UIControlState)state
     UI_APPEARANCE_SELECTOR;
 
+/*
+ A color used as the chip's color for @c state.
```
Comment (jverkoey) [inline components/Chips/src/MDCChipView.h]: I wonder if "background overlay color" might be a more accurate/descriptive term?

**Review by jverkoey — DISMISSED** [review]

File: components/Chips/src/MDCChipView.m (hunk 2 above)
Comment (mohammadcazig) [reply components/Chips/src/MDCChipView.m]: oh yeah, this is needed to have the _chipview shape be the same as the chip. If you think this is not clear I could add a ocmment.

File: components/Chips/src/MDCChipView.h (hunk 3 above)
Comment (mohammadcazig) [reply components/Chips/src/MDCChipView.h]: Hmmmm I don't really wanna change the existing API since we have to deprecate it. But we could plan this in future.

Comment (mohammadcazig) [reply components/Chips/src/MDCChipView.h]: okay I could change the name.

File: components/Chips/src/MDCChipView.m [hunk 4]
```diff
@@ -609,6 +639,14 @@ - (void)layoutSubviews {
         [UIBezierPath bezierPathWithRoundedRect:self.bounds cornerRadius:cornerRadius].CGPath;
   }
 
+  // make sure chipView layer has the same shape the component
+  CAShapeLayer *shapedLayer = (CAShapeLayer *)_backgroundOverlayView.layer.mask;
```
Comment (jverkoey) [inline components/Chips/src/MDCChipView.m]: `backgroundOverlayShapedLayer` to avoid confusion with other shaped layers that might be in use here.

**Review by jverkoey — DISMISSED** [review]

File: components/Chips/src/MDCChipView.m (hunk 4 above)
Comment (mohammadcazig) [reply components/Chips/src/MDCChipView.m]: Agreed.

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
Note: this was rolled back in #3367 as part of the v54.6.0 release.

**Comment by romoore** [comment]
Why did this get rolled-back? The release PR doesn't have any comments relating to the problems that were found.

**Comment by jverkoey** [comment]
Rolled back because we are pursuing a new approach in #3355 and we do not intend to land these APIs.
